mobilecoind-dev-faucet/src/worker.rs

```diff
@@ -570,83 +590,406 @@ impl WorkerTokenState {
         self.known_utxos
             .retain(|key_image, _tracker| output_list_key_images.contains(key_image));
 
-        // Check the queue depth, and decide if we should make a split tx
-        if self.queue_depth.load(Ordering::SeqCst) < target_queue_depth {
-            // Check if we already tried to fix this in the last iteration
-            if let Some(prev_tx) = self.in_flight_split_tx_state.as_ref() {
-                if is_tx_still_in_flight(client, prev_tx, "Split", logger) {
-                    // There is already a fix in-flight, let's do nothing until it lands.
-                    return Ok(());
-                }
+        // Steps 3 and 4 require to consider whether to submit any txs.
+
+        // Get all the "non-target-value" utxos of this token id.
+        let mut non_target_value_utxos: Vec<_> = resp
+            .take_output_list()
+            .into_iter()
+            .filter(|utxo| utxo.token_id == self.token_id && utxo.value != self.target_value)
+            .collect();
+        // Sort in descending order of value
+        non_target_value_utxos.sort_by(|a, b| b.value.cmp(&a.value));
+
+        // Take the MAX_OUTPUTS largest utxos, these will be passed to
+        // "maybe_send_split_txs" for consideration.
+        let top_utxos = &non_target_value_utxos
+            [0..core::cmp::min(MAX_OUTPUTS as usize, non_target_value_utxos.len())];
```

Comment:
Optional/Equivalent: [`Ord::min`](https://doc.rust-lang.org/std/cmp/trait.Ord.html#method.min)
```suggestion
            [0..non_target_value_utxos.len().min(MAX_OUTPUTS_USIZE)];
```

Reply:
TIL that's a thing, nice